agent/src/index.ts

```diff
@@ -488,12 +462,133 @@ export async function createAgent(
         throw new Error("Invalid TEE configuration");
     }
 
-    let goatPlugin: any | undefined;
-    if (getSecret(character, "ALCHEMY_API_KEY")) {
-        goatPlugin = await createGoatPlugin((secret) =>
-            getSecret(character, secret)
-        );
-    }
+    const dynamicPlugins = await loadPlugins(character, [
+        {
+            secrets: ["COINBASE_API_KEY", "COINBASE_PRIVATE_KEY"],
+            importFn: () =>
+                import("@elizaos/plugin-coinbase").then(
+                    (m) => m.coinbaseMassPaymentsPlugin
+                ),
+        },
+        {
+            secrets: ["COINBASE_API_KEY", "COINBASE_PRIVATE_KEY"],
+            importFn: () =>
+                import("@elizaos/plugin-coinbase").then((m) => m.tradePlugin),
+        },
+        {
+            secrets: ["COINBASE_API_KEY", "COINBASE_PRIVATE_KEY"],
+            importFn: () =>
+                import("@elizaos/plugin-coinbase").then(
+                    (m) => m.tokenContractPlugin
+                ),
+        },
+        {
+            secrets: ["COINBASE_API_KEY", "COINBASE_PRIVATE_KEY"],
+            importFn: () =>
+                import("@elizaos/plugin-coinbase").then(
+                    (m) => m.advancedTradePlugin
+                ),
```

Comment:
I was thinking a helper function (my idea of load plugins) which is just `import('@elizaos/plugin-coinbase').then((mod) => [mod.pluginA, modPluginB, modPluginC])`. Then we don't need to import coinbase 3 times.

Reply:
And then to add them you can just spread:

```ts
plugins: [
...myCoinbasePluginsDynamicallyImportedOrNull
]
```

Reply:
ok this final point I think, taking break, will work on that, we getting close 👍🏻

Reply:
This not so simple with diff conditional Coinbase secrets, not all packages rely on same secrets, so can't just import [mod.pluginA, modPluginB, modPluginC] in 1 helper I think, still think need the secret definitions unique to each module, playing with `importModuleWithCache()` solution:

```
const moduleCache: { [key: string]: Promise<Plugin> } = {};

export async function importModuleWithCache(
    moduleName: string
): Promise<Plugin | null> {
    let module = null;
    if (!moduleCache[moduleName]) {
        module = await import(moduleName)
        moduleCache[moduleName] = module;
    }
    return module;
}
``` 

but loses types, tried generics but also bad cause then need to define ` interface CoinBasePlugins`. ugh.

Reply:
solved importCache I believe: https://github.com/elizaOS/eliza/pull/1383/commits/32c7e5a1ad3d5893a806166a4921ca218acb7c0b

Main issue was handling unique secrets, I do like the simple statements of `secrets` and `importFn`, now if duplicate importFn, it uses a cache, so won't import duplicates, but DX is much nicer else would have big nested ternaries and if/else stuffed in array with secrets spread. 

If have better style lemme know.

Reply:
In case that secrets are exact same, uses your suggestion, as array to import: https://github.com/elizaOS/eliza/pull/1383/commits/430dab17023b16a95253ddae952578406cc4a8cf

But case still stands I think best separated if secrets are unique, for webhook, and others.
https://github.com/elizaOS/eliza/pull/1383/files#diff-935219608f7b5ca6c8b8548cfdce88c7d3cdb6bb6d9f9d8df644b364f6557e4eR438-R459